README.md

```diff
@@ -854,10 +854,93 @@ To build an APK to share run (e.g. via Slack), run `npm run android-build`, this
 # Onyx derived values
 Onyx derived values are special Onyx keys which contain values derived from other Onyx values. These are available as a performance optimization, so that if the result of a common computation of Onyx values is needed in many places across the app, the computation can be done only as needed in a centralized location, and then shared across the app. Once created, Onyx derived values are stored and consumed just like any other Onyx value.
 
+## When to use derived values?
+
+1. **Complex Computations Across Multiple Components**
+   - Multiple components need the same computed value from one or more Onyx keys
+   - The computation is expensive (e.g., filtering large arrays, complex object transformations)
+   - The result needs to be cached and shared to avoid redundant calculations
+
+2. **Performance Critical Paths**
+   - The computation appears in frequently rendered components
+   - Profiling shows the same calculation being done repeatedly
+   - The computation involves multiple Onyx dependencies that change independently
+
+3. **Data Aggregation and Transformation**
+   - You need to combine data from multiple Onyx keys into a single, normalized structure
+   - The transformation logic is complex and reusable
+   - The derived data structure is used in multiple places
+
+4. **State-Dependent Calculations**
+   - The value depends on multiple pieces of state that can change independently
+   - The relationship between states is complex (e.g., filtering + sorting + grouping)
+   - Changes in any dependency should trigger a recalculation
+
+## When not to use derived values?
+
+1. **Simple or Local Computations**
+   - The computation is trivial (e.g., simple string manipulation, basic math)
+   - The value is only used in one component
+
+2. **High-Frequency Updates**
+   - The dependencies change very frequently
+   - The derived value would trigger too many re-renders
```

Comment:
do you mind explaining how is this different from the dependencies on their own changing too frequently? wouldn't that be causing unavoidable re-renders? I'm not sure I follow why a derived value is any different. e.g. suppose we have a few components that use dependency A & B and we want them to use instead a computed value C derived from A&B. How is re-renders coming from C worse than re-renders coming from A&B without a derived value?

Reply:
Yeah you are right, it's not different at all, let's remove it

Reply:
I think this needs to be removed too no? 

```
 **High-Frequency Updates**
   - The dependencies change very frequently
```